sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/PayloadSizeGate.java

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.implementation;
+
+import reactor.core.publisher.Flux;
+
+import java.nio.ByteBuffer;
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * This class provides ability to measure if incoming Flux of ByteBuffers is larger than a threshold.
+ * This answers question if volume of data in bytes is larger than threshold.
+ *
+ * The {@link #write(ByteBuffer)} operation buffers incoming ByteBuffers until threshold is crossed.
+ * After that it's pass-through as fact that data volume exceeds threshold is already determined.
+ *
+ * It is incumbent upon the caller to return the buffers after measurement is completed. It is also the caller's
```

Comment:
I think you can delete this line as there's no concept of returning buffers here.

Reply:
deleted